woocommerce/woocommerce
# Product Collection: Migrate `displayLayout` To `supports.layout`

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This pull request replaces the Product Collection block's custom `displayLayout` with [Gutenberg's native `layout` block support](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#layout). This allows us to take advantage of `spacing` support as well as the continued development of `layout`. This migration lets us avoid creating our own CSS styles and controls to handle editing `margin`, `padding`, and `block spacing`.

Closes #43307.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a Product Collection block and edit the layout on the collection block. Test it out!
2. There are new controls for Margin, Padding, and Block Spacing on the collection and template blocks to try out.
3. Create a Product Collection block on `trunk`:
  a. Confirm that the render callback works on this branch with an immigrated block.
  b. Confirm that the migration works and converts different configurations to the new attribute.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
